src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/SplitIntoNestedIfStatementsTests.cs

```diff
@@ -0,0 +1,778 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.SplitOrMergeIfStatements;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.SplitOrMergeIfStatements
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsSplitIntoNestedIfStatements)]
+    public sealed class SplitIntoNestedIfStatementsTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CSharpSplitIntoNestedIfStatementsCodeRefactoringProvider();
+
+        [Theory]
+        [InlineData("a [||]&& b")]
+        [InlineData("a &[||]& b")]
+        [InlineData("a &&[||] b")]
+        [InlineData("a [|&&|] b")]
+        public async Task SplitOnAndOperatorSpans(string condition)
+        {
+            await TestInRegularAndScriptAsync(
+$@"class C
+{{
+    void M(bool a, bool b)
+    {{
+        if ({condition})
+        {{
+        }}
+    }}
+}}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            if (b)
+            {
+            }
+        }
+    }
+}");
+        }
+
+        [Theory]
+        [InlineData("a [|&|]& b")]
+        [InlineData("a[| &&|] b")]
+        [InlineData("a[||] && b")]
+        public async Task NotSplitOnAndOperatorSpans(string condition)
+        {
+            await TestMissingInRegularAndScriptAsync(
+$@"class C
+{{
+    void M(bool a, bool b)
+    {{
+        if ({condition})
+        {{
+        }}
+    }}
+}}");
+        }
+
+        [Fact]
+        public async Task NotSplitOnIfKeyword()
```

Comment:
Will do, but then I'd probably prefer if it still worked even if there are multiple operators and just picked maybe the last once (since that is the actual root expression... in other cases we are relying on associativity of the operators). Otherwise users would get used to invoking this on the `if` but then be surprised when it's not offered.

Reply:
There is a concern of noisiness. Right now I put this feature above invert if in the lightbulb menu because invoking on the operator is really specific and doing this seemed appropriate. But if we offer this on the `if` keyword, that would have to be changed because this will show up in a lot of if statements and we don't want muscle memory for invert if to stop working.

Reply:
I didn't care about putting this on the `if` keyword and having this show up less often that the merging refactorings because generally the merging refactorings simplify code, whereas this is expanding it (we don't want that to be too noisy).

Reply:
Actually now I'm convincing myself to not dot that :smile:

Reply:
Given that the splitting refactorings expand code and make it more complex, offering this on the `if` keyword might be too common and it might seem like this is a suggestion rather than a refactoring.

Reply:
If the concern is breaking invert if, just put it after invert if? Unless we had strong indications of which we think is going to be used more, it might generally be easiser to put invert if first just so the list's order is generally more stable. As it stands today, you're essentially *moving* invert if in semi-rare circumstances that might make the list ordering feel more flaky as it is.

Reply:
I admit I don't have a good answer for this myself; this might be something for us to take to a design meeting.

Reply:
Having this below if in the case I'm directly at the operator seemed a little annoying. The current context is much closer to the operator itself than an action like negating the whole if statement.

Reply:
It's annoying if you're trying to invoke yours. However you just annoyed everybody who sticks their caret somewhere inside of the condition and wants to invoke Invert If. Who wins? :smile:

Reply:
(sticks their caret and incidentally landed on a conditional. Basically, is caret on conditional enough to truly signal intent there?)

Reply:
@jasonmalinowski "Invert if" is already way down in that case:
![image](https://user-images.githubusercontent.com/11444821/50366130-40a0b880-0578-11e9-8962-b9b2eada4f4e.png)

which seems appropriate to me. Items are usually ordered in terms of locality.